dttools/src/priority_queue.h

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright (C) 2024 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#ifndef PRIORITY_QUEUE_H
+#define PRIORITY_QUEUE_H
+
+/** 
+@file priority_queue.h
+A general purpose priority queue.
+This priority queue module is implemented as a complete binary heap that manages elements with associated priorities.
+Each element in the priority queue has a priority, and the queue ensures that the element with the highest priority
+can be accessed in constant time.
```

Comment:
Is that the only guarantee offered here, or does it also guarantee that iteration is performed in order?